cirq-core/cirq/sim/simulator_test.py

```diff
@@ -389,21 +389,6 @@ def test_simulation_trial_result_qubit_map():
     assert result.qubit_map == {q[0]: 0, q[1]: 1}
 
 
-def test_verify_unique_measurement_keys():
```

Comment:
Actually, what *is* the expectation when sampling and repeated keys? The test now slightly changes the error message and that's all.

Reply:
(Update: I went ahead and updated the code to return the old error message, which was more informative. Whether this is the desired behavior for samplers remains an open question).

Reply:
`cirq.sample` returns a `Result` from what I can see - I think the expectation should be the same as for `simulator.run` (i.e. return a Result with multiple measurements per key).

This may also need to wait on #4555, or be applied after both of these PRs go in.